test/evp_test.c

```diff
@@ -473,15 +473,15 @@ static int digest_test_run(EVP_TEST *t)
         xof |= 1;
         *p++ = OSSL_PARAM_construct_size_t(OSSL_DIGEST_PARAM_XOFLEN,
                                            &expected->output_len);
-    }
-    if (expected->digest_size > 0) {
+    } else if (expected->digest_size > 0) {
         *p++ = OSSL_PARAM_construct_size_t(OSSL_DIGEST_PARAM_SIZE,
                                            &expected->digest_size);
     }
     if (expected->pad_type > 0)
         *p++ = OSSL_PARAM_construct_int(OSSL_DIGEST_PARAM_PAD_TYPE,
                                         &expected->pad_type);
     *p++ = OSSL_PARAM_construct_end();
```

Comment:
I would prefer in the last *p++ was just *p =

Reply:
Fixed, you get the joy of repairing this supposed "mistake" everywhere else however.